src/Mod/PartDesign/App/FeatureExtrude.cpp

```diff
@@ -480,18 +480,36 @@ App::DocumentObjectExecReturn* FeatureExtrude::buildExtrusion(ExtrudeOptions opt
 
     // Validate parameters
     double L = Length.getValue();
-    if ((method == "Length") && (L < Precision::Confusion())) {
-        return new App::DocumentObjectExecReturn(
-            QT_TRANSLATE_NOOP("Exception", "Length too small"));
+    if (method == "Length") {
+        if (std::abs(L) < Precision::Confusion()) {
+            if (addSubType == Type::Additive)
+                return new App::DocumentObjectExecReturn(
+                    QT_TRANSLATE_NOOP("Exception", "Cannot create a pad with a height of zero."));
+            else
+                return new App::DocumentObjectExecReturn(
+                    QT_TRANSLATE_NOOP("Exception", "Cannot create a pocket with a depth of zero."));
+        }
     }
     double L2 = 0;
-    if ((method == "TwoLengths")) {
+    if (method == "TwoLengths") {
         L2 = Length2.getValue();
-        if (std::abs(L2) < Precision::Confusion()) {
-            return new App::DocumentObjectExecReturn(
-                QT_TRANSLATE_NOOP("Exception", "Second length too small"));
+        if (std::abs(L2) < Precision::Confusion() && std::abs(L) < Precision::Confusion()) {
+            if (addSubType == Type::Additive)
+				return new App::DocumentObjectExecReturn(
+                    QT_TRANSLATE_NOOP("Exception", "Length2 too small"));
+			else
+				return new App::DocumentObjectExecReturn(
+                    QT_TRANSLATE_NOOP("Exception", "Length2 too small"));
         }
     }
+    if (std::abs(L + L2) < Precision::Confusion()) {
```

Comment:
Here you are using L2 but this variable is not initialized in this block of code I think. Because `L2 = Length2.getValue();` is within the previous if (method == "TwoLengths") block. So here L2 is always 0 as it's defined before the IF